solver/src/main/java/Smt2Writer.java

```diff
@@ -28,7 +24,7 @@ public class Smt2Writer  {
     private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(Smt2Writer.class.getName());
 
     // The variables that solve the constraint
-    private final Set<String> variables = new HashSet<>();
+    private final Map<String, String> variables = new HashMap<>();
```

Comment:
everytime you have maps as fields, in comments always specify what is the 'key' and 'value' entries, especially when usig primitives such as String